libyara/modules/dex/dex.c

```diff
@@ -418,7 +418,7 @@ static int64_t dex_get_integer(
     const char* pattern,
     int64_t index)
 {
-  if (index == YR_UNDEFINED || index < 0)
+  if (index < 0)
```

Comment:
I wouldn't rely on`YR_UNDEFINED` being < 0 as it makes the intention less obvious. `if (index == YR_UNDEFINED || index < 0)` makes clear that index could be undefined and what to do in that case, while `if (index < 0)` does not.